### PR TITLE
add execution contexts

### PR DESCRIPTION
**Pull request by mikeclimbrock**

### CODE REVIEW

File: api/app/utils/Google.scala [hunk 1]
```diff
@@ -75,7 +78,9 @@ class Google @javax.inject.Inject() (
     Timezones.find(tz.getID())
   }
 
-  def getLocationsByAddress(address: String): Seq[Address] = {
+  def getLocationsByAddress(address: String)(
+    implicit ec: ExecutionContext
```
Comment (mikeclimbrock) [inline api/app/utils/Google.scala]: not sure this is doing anything yet

File: api/app/controllers/Helpers.scala [hunk 2]
```diff
@@ -32,29 +36,33 @@ class Helpers @javax.inject.Inject() (
...
-            Right(Nil)
+            Future { Right(Nil) }
           }
 
           case Some(c) => {
-            Right(
-              Seq(
-                Address(
-                  country = Some(c.iso31663)
+            Future {
```
Comment (chrhicks) [inline api/app/controllers/Helpers.scala]: I think you want `Future.successful` here. See: https://groups.google.com/forum/#!topic/scala-user/nBtj9XW2NfQ (Sam Halliday reply)

File: api/app/controllers/Helpers.scala [hunk 3]
```diff
@@ -63,31 +71,33 @@ class Helpers @javax.inject.Inject() (
...
+            }
           }
         }
       }
           
       case (_, _, Some(i)) => {
         MaxMind.getByIp(i) match {
-          case Some(address) => Right(Seq(address))
-          case None => Right(Nil)
+          case Some(address) => Future { Right(Seq(address)) }
```
Comment (chrhicks) [inline api/app/controllers/Helpers.scala]: I think you want `Future.successful` here.  for the three here as well.

File: api/app/controllers/Addresses.scala [hunk 4]
```diff
@@ -1,54 +1,52 @@
...
-              sys.error(s"Error in address verification: $errors")
-            }
-
-            case Right(locations) => {
-              Ok(Json.toJson(AddressVerifier(address, locations)))
-            }
+    val address = request.body.as[Address]
+    AddressVerifier.toText(address) match {
+      case None => {
+        Future { UnprocessableEntity(Json.toJson(Validation.error("Address to verify cannot be empty"))) }
```
Comment (chrhicks) [inline api/app/controllers/Addresses.scala]: I think you want `Future.successful` here.

File: api/app/controllers/Helpers.scala [hunk 5]
```diff
@@ -32,29 +36,33 @@ class Helpers @javax.inject.Inject() (
...
-  ): Either[Unit, Seq[Address]] = {
+  )(
+    implicit ec: ExecutionContext
+  ): Future[Either[Unit, Seq[Address]]] = {
     (country, address, ip) match {
       case (Some(code), _, _) => {
         Countries.find(code) match {
           case None => {
-            Right(Nil)
+            Future { Right(Nil) }
```
Comment (chrhicks) [inline api/app/controllers/Helpers.scala]: I think you want `Future.successful` here.

File: api/app/controllers/Helpers.scala [hunk 6]
```diff
@@ -63,31 +71,33 @@ class Helpers @javax.inject.Inject() (
         // Special case to enable specifying just a country code in the address line
         Countries.find(a) match {
           case Some(c) => {
-            Right(
-              Seq(
-                Address(
-                  country = Some(c.iso31663)
+            Future {
```
Comment (chrhicks) [inline api/app/controllers/Helpers.scala]: I think you want `Future.successful` here.

**Pull request merged**